admin/helpers/table-sorting.js

```diff
@@ -0,0 +1,50 @@
+const tableSorting = {
+
+  /**
+   * Sort a list of objects based on a field
+   * @param {Array} tableData
+   * @param {String} field
+   * @param {Boolean} asc
+   * @returns {Array} sorted array data
+   */
+  applySorting: function (tableData, field, asc = true) {
+    return tableData.sort((a, b) => this.comparer(a[field], b[field], asc))
+  },
+
+  comparer: function (v1, v2, asc) {
+    return this.isNumericValue(v1) && this.isNumericValue(v2)
+      ? this.getNumberComparisonResult(v1, v2, asc) : this.getStringComparisonResult(v1, v2, asc)
+  },
+
+  isNumericValue: function (v) {
+    const numericOnlyPattern = /^\d+$/
+    const numericOnlyRegExp = new RegExp(numericOnlyPattern)
```

Comment:
This Regexp only needs to be generated once, rather than every time.  It's static.

Reply:
@jon-shipley definitely, as a matter of fact I'll have to reflect that change on the client side sorting too.

Reply:
done